processor/src/main/java/org/mapstruct/ap/internal/model/MapMappingMethod.java

```diff
@@ -210,15 +208,26 @@ private Assignment forgeMapping(SourceRHS sourceRHS, Type sourceType, Type targe
                 )
             );
 
+            BeanMappingMethod forgedMappingMethod = new BeanMappingMethod.Builder()
```

Comment:
I see this coming back several times. Can't we put the method on the context and parameterize?